### PR TITLE
Refactor interop usage in DirectoryServices

### PR DESCRIPTION
**Pull request by mthalman**
This is a partial implementation to more closely conform to the [interop guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/interop-guidelines.md).

The goals here were the following:
* Clean up the ExternDlls class to only contain strings for those assemblies that are used by and unique to System.DirectoryServices and System.DirectoryServices.AccountManagement.
* Remove hardcoded library names in `[DllImport]`.
* Use the common Interop.Libraries class when possible.

### CODE REVIEW

**Comment by mthalman** [comment]
FYI: The motivation for these changes is to remove cruft and provide a clearer picture of what dependencies libraries have.  This is to support the work that is going on for https://github.com/dotnet/core/issues/5651.

File: src/libraries/System.DirectoryServices/src/System/DirectoryServices/ExternDll.cs [hunk 1]
```diff
@@ -6,45 +6,7 @@ namespace System
...
-        public const string Crypt32 = "crypt32.dll";
-
-        // system.data specific
-        internal const string Odbc32 = "odbc32.dll";
-        internal const string SNI = "System.Data.dll";
-
-        // system.data.oracleclient specific
-        internal const string OciDll = "oci.dll";
-        internal const string OraMtsDll = "oramts.dll";
+        public const string Dnsapi = "dnsapi.dll";
```
Comment (eerhardt) [inline src/libraries/System.DirectoryServices/src/System/DirectoryServices/ExternDll.cs:9]: Can these move to [https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/Interop/Windows/Interop.Libraries.cs](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/Interop/Windows/Interop.Libraries.cs)?

Comment (eerhardt) [reply src/libraries/System.DirectoryServices/src/System/DirectoryServices/ExternDll.cs:9]: See https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/interop-guidelines.md#file-organization

Comment (mthalman) [reply src/libraries/System.DirectoryServices/src/System/DirectoryServices/ExternDll.cs:9]: I could.  I wasn't intending on changing things to fully comply with the interop guidelines but rather to just get closer to them.  But just moving these constants to common seems simple and reasonable enough to do.

Comment (mthalman) [reply src/libraries/System.DirectoryServices/src/System/DirectoryServices/ExternDll.cs]: Fixed

**Review by eerhardt — APPROVED** [review]
LGTM. Nice work cleaning this up!

fyi - @buyaa-n @joperezr @krwq

File: src/libraries/System.DirectoryServices/src/Interop/SafeNativeMethods.cs [hunk 2]
```diff
@@ -25,10 +25,10 @@ public const int
             ERROR_MORE_DATA = 234,
             ERROR_SUCCESS = 0;
 
-        [DllImport(ExternDll.Activeds, CharSet = CharSet.Unicode)]
+        [DllImport(global::Interop.Libraries.Activeds, CharSet = CharSet.Unicode)]
         public static extern unsafe int ADsGetLastError(out int error, char* errorBuffer, int errorBufferLength, char* nameBuffer, int nameBufferLength);
 
-        [DllImport(ExternDll.Activeds, CharSet = CharSet.Unicode)]
+        [DllImport(global::Interop.Libraries.Activeds, CharSet = CharSet.Unicode)]
```
Comment (stephentoub) [inline src/libraries/System.DirectoryServices/src/Interop/SafeNativeMethods.cs:31]: Why are all these `global::`s necessary?  Is there some kind of naming conflict this is working around?

Comment (eerhardt) [reply src/libraries/System.DirectoryServices/src/Interop/SafeNativeMethods.cs:31]: > Is there some kind of naming conflict this is working around?

Yes, there is an `Interop` namespace in this project.

Comment (stephentoub) [reply src/libraries/System.DirectoryServices/src/Interop/SafeNativeMethods.cs:31]: Ah, I see, the `System.DirectoryServices.Interop` namespace.  Presumably once we clean up the rest of the interop we can remove the "global::".

**Review by stephentoub — APPROVED** [review]
Thanks for the clean up.

**Review by joperezr — APPROVED** [review]
Thanks so much for taking care of this @mthalman!

**Pull request merged**